zio/zngio/scanner.go

```diff
@@ -59,6 +133,16 @@ func (s *zngScanner) Pull() (zbuf.Batch, error) {
 }
 
 func (s *zngScanner) scanUncompressed() ([]*zng.Record, error) {
+	if s.stringFinder != nil {
+		buf := s.reader.uncompressed.Bytes()
+		if s.stringFinder.next(byteconv.UnsafeString(buf)) == -1 {
+			// We know s.reader.uncompressed cannot contain any
+			// records matching s.filter.
+			s.reader.uncompressed = nil
+			// xxx stats
```

Comment:
Efficiently updating scanner.ScannerStats.RecordsRead probably requires that we change the ZNG spec so that a compressed value message block stores the number of records it contains. Do we need to do that?

Reply:
Since we BytesRead & RecordsRead, we could just interpret RecordsRead as "instantiated records needed for this operation". In which case, we wouldn't need to update RecordsRead here.

Reply:
Works for me.

Reply:
I think we are going to want to have records read reflect the number of records read from disk. The number of instantiated records is a gray area between records read and records match. I am skeptical of its utility at that point. This would mean I am in favor of storing the number of records in a compressed block.

Could be done down the road, but for now I would just assume drop RecordsRead from scanner stats for the time being if not done in this pr.

Reply:
I'd like to move forward with what's here (a skipped compressed message value block doesn't affect RecordsRead) and revisit the question for #1010.